mongo/index_view.go

```diff
@@ -354,7 +374,17 @@ func (iv IndexView) drop(ctx context.Context, name string, opts ...*options.Drop
 	return res, nil
 }
 
-// DropOne drops the index with the given name from the collection.
+// DropOne executes a dropIndexes operation to drop an index on the collection. If the operation succeeds, this returns
+// a BSON document in the form {nIndexesWas: <int32>}. The "nIndexesWas" field in the response contains the number of
```

Comment:
I'm assuming this is just what the server returns?  Is it the only field?

I'm not sure we should try to guarantee anything about this in the public API specification, as the server has been known to change these fields from release to release.

Reply:
This is what the server returns, but we actually unmarshal it into a concrete result type in the `operation` package and then return as raw BSON here. This was done because the return type for this function was already `bson.Raw` when we implemented the new core API. It's something I'd like to change in v2.0 of the driver when we get to that.

Given all of this, I'm not sure this is any different than result types we return. If the server were to ever change what it returns, we could still transform it correctly because it goes through a BSON -> Go -> BSON round trip.